### ISSUE TITLE
Duplicate hostPorts Rejected Despite Disparate Protocols

### ISSUE THREAD

**Issue by eatnumber1** [GoogleCloudPlatform/kubernetes#8372]
Pods which use `hostPort` to bind both TCP and UDP sockets to the same port number are rejected with the error:

```
ERROR: Error: Pod "foo" is invalid: spec.containers[0].port: duplicate value '2049'
```

An example pod follows.

```yaml
apiVersion: v1beta3
kind: Pod
metadata:
  name: foo
spec:
  containers:
    - name: foo
      image: foo/bar
      ports:
        - name: foo-tcp
          containerPort: 2049
          protocol: tcp
          hostPort: 2049
        - name: foo-udp
          containerPort: 2049
          protocol: udp
          hostPort: 2049
```

**Comment by bprashanth** [comment]
Looks like we don't check protocol
https://github.com/GoogleCloudPlatform/kubernetes/blob/bc9b6e93e3ad9466ac3ba24d2d2ad80f04060225/pkg/api/validation/validation.go#L701
@thockin

**Comment by yujuhong** [comment]
The issue was discussed before in #5874, which was closed due to it being low priority.

**Comment by bprashanth** [comment]
Hmm, that's 2 pods and the scheduler, this is 2 containers in the same pod and validation at create time. Maybe we should just be more clear about where we stand with hostPort.

**Comment by eatnumber1** [comment]
Posting here what I posted in the other bug:

This can't be worked around when running NFS from a pod. Linux does not allow the NFS port (2049) to be changed, and the Portmap port (111) also cannot be changed. Additionally, you can't separate the UDP from the TCP ports.

**Comment by thockin** [comment]
Why does NFS need host ports?

**Comment by thockin-cc** [comment]
Fix is in #8490
